spring-data-couchbase/src/main/java/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.java

```diff
@@ -379,11 +400,28 @@ protected boolean autoIndexCreation() {
 	 * and {@link #couchbaseMappingContext(CustomConversions)}. Returns an empty {@link CustomConversions} instance by
 	 * default.
 	 *
+	 * @param cryptoManagerOptional optional cryptoManager. Make varargs for backwards compatibility.
 	 * @return must not be {@literal null}.
 	 */
 	@Bean(name = BeanNames.COUCHBASE_CUSTOM_CONVERSIONS)
-	public CustomConversions customConversions() {
-		return new CouchbaseCustomConversions(Collections.emptyList());
+	public CustomConversions customConversions(CryptoManager... cryptoManagerOptional) {
+		assert (cryptoManagerOptional == null || cryptoManagerOptional.length <= 1);
```

Comment:
What do you think of these alternatives stolen from StackOverflow?

```
public CustomConversions customConversions(
    Optional<CryptoManager> cryptoManagerOptional
)
```

Or this:

```
public CustomConversions customConversions(
    @Autowired(required = false) CryptoManager cryptoManager
)
```

Reference: https://stackoverflow.com/questions/46704950/how-to-define-optional-parameters-dependencies-in-the-bean-method-in-spring

Reply:
The first one changes a public API so is not possible.
The second one - cryptoManager is not a bean and therefore cannot be instantiated with @Autowired. I also have doubts about using @Autowired in the configuration class.  And also bean-ifying things that don't absolutely need to be beans.

Reply:
Changing `foo()` to `foo(T...)` is a source-compatible change, but I suspect it's not binary-compatible. (Maybe binary compatibility is not a concern?)